core/src/test/java/nl/nn/adapterframework/senders/ShadowSenderTest.java

```diff
@@ -211,6 +212,60 @@ public void testResultSenderResultWith3ShadowSenders() throws Exception {
 		}
 	}
 
+	@Test
+	public void testResultSenderResultWith3ShadowSendersAsync() throws Exception {
+		// Arrange
+		sender.registerSender(new TestSender("shadowSenderWithDelay1"));
+		sender.registerSender(new TestSender("shadowSenderWithDelay2"));
+		sender.registerSender(new TestSender("shadowSenderWithDelay3"));
+		((ShadowSender)sender).setWaitForShadowsToFinish(false);
+
+		sender.configure();
+		sender.open();
+
+		Message inputMessage = MessageTestUtils.getNonRepeatableMessage(MessageTestUtils.MessageType.CHARACTER_UTF8);
+
+		// Act
+		inputMessage.closeOnCloseOf(session, sender);
+		String result = sender.sendMessageOrThrow(inputMessage, session).asString();
+		session.close();
+
+		// Assert
+		assertEquals(ORIGINAL_SENDER_RESULT, result);
+
+		Thread.sleep(3000); // wait for results to be collected in the background
```

Comment:
Is it easy to convert this to use `await()` method, so this unittest perhaps runs faster when possible?

Reply:
Not sure. Will look at it.